src/libYARP_OS/src/DgramTwoWayStream.cpp

```diff
@@ -222,7 +222,9 @@ void DgramTwoWayStream::allocate(int readSize, int writeSize) {
                 int result = dgram->get_option(SOL_SOCKET, SO_RCVBUF, &_read_size, &len);
                 if (result < 0) {
                     YARP_ERROR(Logger::get(), ConstString("Failed to read buffer size from RCVBUF socket with error: ") +
-                               ConstString(strerror(errno)));
+                               ConstString(strerror(errno)) +
+                               ConstString(". Setting read buffer size to UDP_MAX_DATAGRAM_SIZE."));
+                    _read_size = UDP_MAX_DATAGRAM_SIZE;
```

Comment:
The only doubt is about this assignment, should be better to put it to 0?? Anyway it is quite impossible that `result < 0` so it is not a big deal.

Reply:
The ratio here is that `_read_size` is used afterwards to allocate memory.
Allocating zero memory will lead to an unusable buffer, while the client code expects something usable, but the API does not allow to notify the user that something went wrong.

Few lines above, at 211, a similar solution has been adopted for `_write_size`
```
if (_write_size > UDP_MAX_DATAGRAM_SIZE || _write_size < 0)
{
         _write_size = UDP_MAX_DATAGRAM_SIZE;
}
```
so I think it is safe to adopt for `_read_size` as well. Anyway I agree with you that it is a corner case that should never happen (and I believe never was).

Reply:
All this stuff has been added by me in this [commit](https://github.com/robotology/yarp/commit/ed59342b6e96c973c80e7ba7cc6ce375b5df8a33) and I was uncertain how handle this error 😅 
BTW I agree to `_read_size = UDP_MAX_DATAGRAM_SIZE`  for symmetry 👍